cinema/permissions.py

```diff
@@ -0,0 +1,16 @@
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+    """
+    Permission allows administrators full access (read and write),
+    while regular authenticated users have read-only access
+    """
+
+    def has_permission(self, request, view):
+        if request.method in SAFE_METHODS:
+            return bool(
+                request.method
+                in SAFE_METHODS

```

Comment:
The check `request.method in SAFE_METHODS` is redundant here because it is already checked in the `if` statement on line 11. You can simplify the logic by removing the repeated check within the `bool()` function.